lib/nanoc/base/checksummer.rb

```diff
@@ -18,23 +18,28 @@ def calc(obj)
 
       private
 
-      def update(obj, digest)
+      def update(obj, digest, visited = Set.new)
         digest.update(obj.class.to_s)
 
+        if visited.include?(obj)
+          digest.update('recur')
```

Comment:
if visited would be a hash then we could store also the digest value and set it here, would it be useful at all?